include/simdjson/generic/ondemand/json_iterator-inl.h

```diff
@@ -92,7 +102,7 @@ simdjson_warn_unused simdjson_really_inline error_code json_iterator::skip_child
   // Now that we've considered the first value, we only increment/decrement for arrays/objects
   auto end = &parser->implementation->structural_indexes[parser->implementation->n_structural_indexes];
   while (token.index <= end) {
-    switch (*advance()) {
+    switch (*return_current_and_advance()) {
```

Comment:
While advance() sounds different from next() to me, obviously it doesn't to everyone, so this is fine :)

I believe I picked this return_current_and_advance method (i++) over a simple next() (++i) mainly because I read there's an addressing mode that can get you the value and post-increment all in one go. It also makes it a little quicker to peek at the next token or get the length of the current token (since getting the token length involves subtracting *next from *current).

Anyway, this is good :) More explicit is better.